src/viam/ur/module/ur_arm.cpp

```diff
@@ -894,6 +903,18 @@ URArm::UrDriverStatus URArm::read_joint_keep_alive_(bool log) {
                 current_state_->driver->stopPrimaryClientCommunication();
                 current_state_->driver->startPrimaryClientCommunication();
 
+                std::string robot_mode;
+                if (!current_state_->dashboard->commandRobotMode(robot_mode)) {
+                    throw std::runtime_error("read_joint_keep_alive dashboard->commandRobotMode() failed to retrieve robot mode");
+                }
+
+                // if the robot is not running, we want to restart the arm fully to make sure we are ready
+                // we will cheat by going into the estop code
+                if (robot_mode.find(urcl::robotModeString(urcl::RobotMode::RUNNING)) == std::string::npos) {
+                    // we are not ready to run, go into the estop code to try to be ready
+                    current_state_->estop.store(true);
```

Comment:
how does setting a bool make the robot try to be ready? is this depending on `read_joint_keepalive` thread to restart underlying connections?

Reply:
yeah, in the [handle estop chunk of code](https://github.com/viam-modules/universal-robots/blob/980e125ae93887d746e3dc51981b99967c3526f0/src/viam/ur/module/ur_arm.cpp#L971-L1000) we go through the process of ensuring the arm is turned on and sending the control script that we want to use. 

rather than redo all of that here I cheat by setting the flag

Reply:
if status is empty or not normal then the arm is currently estopped, so we don't want to turn the arm back on

Reply:
https://viam.atlassian.net/browse/RSDK-11279